contracts/package.json

```diff
@@ -20,7 +20,10 @@
   "author": "James Zaki",
   "license": "MIT",
   "dependencies": {
-    "@openzeppelin/contracts": "^4.7.3"
+    "@openzeppelin/contracts": "^4.7.3",
+    "dotenv": "^16.0.3",
+    "ethers": "^5.7.2",
+    "hardhat": "^2.12.1"
```

Comment:
Hmm. I did this to resolve the lint warnings.

It sounds like you are targeting the use-case where people list bls-wallet in their package.json in order to use the contracts directly.

It's a bit weird because you're not supposed to use `devDependencies` in your core/non-test code. This is why the linter is complaining. I suppose there are three levels of distinction in this project:
1. Solidity code
2. Deployment code
3. Test code

Maybe we could tell the linter somehow that deployment code isn't 'core', like it should treat it more like test code.

Alternatively, it might be more technically correct to make an adjustment to only include the solidity dependencies if/when we decide to publish an npm module for importing our contracts..

Reply:
I looked into those lint warnings awhile back, and did not find a satisfactory answer that didn't involve turning off that lint rule or manually listing packages to ignore.

In the short/medium term I don't think it will matter much. But I think
> Maybe we could tell the linter somehow that deployment code isn't 'core', like it should treat it more like test code.

would be a better way to go.

> Alternatively, it might be more technically correct to make an adjustment to only include the solidity dependencies if/when we decide to publish an npm module for importing our contracts.

If we did this it would also matter more (package bloat), but you can directly import contracts/libs via url so I think there is less of a need for that unless we are also including deployment scripts in that package. Then this would stay as is per your changes.